### DISCUSSION TITLE
Mesh collision seams fail when missing bottom face

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#282] [Q&A] [answered]
Hi,

Currently, I want to define a box shape using triangle mesh. However, there are only five faces are selected without the bottom one. Even the simple box shape collision failed colliding with such mesh shape.
By common sensor, this mesh should be stable standing on the box. I export the ovd and hope to get some explanation for it. Do I need thicken the shape or any other parameters to adjust?

Thank you. 

![image](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/3730f533-1320-417e-81b8-982cd7849da9)
[MeshBoxWithoutBottomFace.zip](https://github.com/NVIDIA-Omniverse/PhysX/files/15291451/MeshBoxWithoutBottomFace.zip)

**Comment by vreutskyy** [comment]
Hi @yananZZZZ 
And this triangle mesh, does it have SDF? As I'm aware, PhysX doesn't support dynamic triangle mesh colliders without SDF.

**Reply by yananZZZZ** [reply]
Hi Victor,
Yes, I aware that SDF is a prerequisite for dynamic mesh, but this mesh doesn't work under GPU with SDF neither. 

One more interesting thing, I notice for other mesh with good tessellation. Even I didn't enable GPU, it can collide with box or other primary collision shape. What's I do is I enable the "buildGPUData" in the cook parameter, but I didn't set the SDF parameters in the mesh description.  I know it might not a good way, but somehow it works just fine. I suspect the code is reroute to some other brunch. Hope to get some input from your side.

thank you.

**Reply by vreutskyy** [reply]
Were there any error messages? Like, it might fail to build SDF because the mesh is open.

**Reply by yananZZZZ** [reply]
Hi Victor,

Indeed, there is a warning "SDF creation: Input mesh is not watertight. The SDF will try to close the holes." caught by our log. Is this indicate the mesh won't work? How should we handle such open mesh to make it work?

Thank you

**Reply by vreutskyy** [reply]
Hi @yananZZZZ 
Can you use a closed mesh here? SDF builder needs the surface of this mesh to be continuous to voxelize it. This means the walls of this crate should have some thickness.
You could also make this crate out of 5 thin boxes, which would be even better.
Or is there some other idea behind this why you want to use a triangle mesh for a dynamic body's collision?

**Reply by vreutskyy** [reply, marked as answer]
Hi @yananZZZZ 
I made more testing and it seems this particular case isn't related to SDF. SDF isn't used for TriangleMesh-Box collision.
The problem is that in PhysX all meshes are single-sided, it means to collide properly your 5-faces cube should have 5 more faces defining its interior. It should work even if these inner faces are coplanar with their outer counterparts and the walls thickness is 0.

**Reply by yananZZZZ** [reply]
Hi Viktor,
Thank you for the reply and it makes sense why some of our mesh collisions work even without GPU on. One more thing is there is a PxMeshGeometryFlag::eDOUBLE_SIDED but is only used for raycast and sweep. Is there any chance that this can be expanded to collision behavior?
Anyhow, I'm happy with the explanation now, hope the section of mesh collision in the documentation can add more details.

**Reply by vreutskyy** [reply]
I asked the guys and they say it's very old story from the times before CCD was introduced. It's unlikely it will change any time soon. And they suggest the same thing - duplicate the faces and flip the normals to have double-sided collision 🤷‍♂️
